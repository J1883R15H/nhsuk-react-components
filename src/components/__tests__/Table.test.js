import React from 'react';
import { shallow } from 'enzyme';
import Table from '../table';

describe('Table', () => {
  describe('Header', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(
        <Table.Header rows={['Skin symptoms', 'Possible cause']} />
      );
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('renders its children', () => {
      const shallowMount = shallow(
        <Table.Header rows={['Skin symptoms', 'Possible cause']} />
      );
      expect(
        shallowMount.containsMatchingElement(
          <th className="nhsuk-table__header" scope="col">
            Skin symptoms
          </th>
        )
      ).toBeTruthy();
      expect(
        shallowMount.containsMatchingElement(
          <th className="nhsuk-table__header" scope="col">
            Possible cause
          </th>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('Row', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(
        <Table.Row
          items={['Blisters on lips or around the mouth', 'cold sores']}
        />
      );
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('properly renders items', () => {
      const shallowMount = shallow(
        <Table.Row
          items={['Blisters on lips or around the mouth', 'cold sores']}
        />
      );
      expect(
        shallowMount.containsMatchingElement(
          <td className="nhsuk-table__cell">
            Blisters on lips or around the mouth
          </td>
        )
      ).toBeTruthy();
      expect(
        shallowMount.containsMatchingElement(
          <td className="nhsuk-table__cell">cold sores</td>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('Body', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(
        <Table.Body>
          <Table.Row />
        </Table.Body>
      );
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
  });
  describe('TablePanel', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<Table.Panel />);
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('renders heading', () => {
      const shallowMount = shallow(<Table.Panel heading="test" />);
      expect(
        shallowMount.containsMatchingElement(
          <h3 className="nhsuk-table__heading-tab">test</h3>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
  describe('NormalTable', () => {
    it('matches snapshot', () => {
      const shallowMount = shallow(<Table />);
      expect(shallowMount).toMatchSnapshot();
      shallowMount.unmount();
    });
    it('renders caption', () => {
      const shallowMount = shallow(<Table caption="Caption" />);
      expect(
        shallowMount.containsMatchingElement(
          <caption className="nhsuk-table__caption">Caption</caption>
        )
      ).toBeTruthy();
      shallowMount.unmount();
    });
  });
});
