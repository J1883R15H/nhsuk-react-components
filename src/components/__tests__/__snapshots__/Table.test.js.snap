// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Body matches snapshot 1`] = `
<tbody
  className="nhsuk-table__body "
  style={Object {}}
>
  <TableRow
    className=""
    style={Object {}}
  />
</tbody>
`;

exports[`Table Header matches snapshot 1`] = `
<thead
  className="nhsuk-table__head "
  style={Object {}}
>
  <tr
    className="nhsuk-table__row"
  >
    <th
      className="nhsuk-table__header"
      key="Skin symptoms"
      scope="col"
    >
      Skin symptoms
    </th>
    <th
      className="nhsuk-table__header"
      key="Possible cause"
      scope="col"
    >
      Possible cause
    </th>
  </tr>
</thead>
`;

exports[`Table NormalTable matches snapshot 1`] = `
<div
  className="nhsuk-table-responsive"
>
  <table
    className="nhsuk-table "
    style={Object {}}
  />
</div>
`;

exports[`Table Row matches snapshot 1`] = `
<tr
  className="nhsuk-table__row "
  style={Object {}}
>
  <td
    className="nhsuk-table__cell"
    key="Blisters on lips or around the mouth"
  >
    Blisters on lips or around the mouth
  </td>
  <td
    className="nhsuk-table__cell"
    key="cold sores"
  >
    cold sores
  </td>
</tr>
`;

exports[`Table TablePanel matches snapshot 1`] = `
<div
  className="nhsuk-table__panel-with-heading-tab "
  style={Object {}}
/>
`;
